src/Docker/Client/Types.hs

```diff
@@ -1391,7 +1392,9 @@ instance FromJSON ContainerConfig where
         image <- o .: "Image"
         volumes <- o .: "Volumes"
         workingDir <- o .:? "WorkingDir"
-        entrypoint <- o .:? "Entrypoint"
+        entrypoint <- asum [ o .:? "Entrypoint" .!= []
+                           , (:[]) <$> o .: "Entrypoint"
```

Comment:
I think this line will always cause parsing to fail when `Entrypoint` isn't a string. Maybe something like the following? 

    (:[]) <$> o .:? "Entrypoint" .!= []

Could you include test cases for parsing each version (null, string, string list)?